test/plugin_functional/plugins/kbn_tp_embeddable_explorer/public/np_ready/public/app/dashboard_container_example.tsx

```diff
@@ -86,7 +86,7 @@ export class DashboardContainerExample extends React.Component<Props, State> {
   }
 
   public switchViewMode = () => {
-    this.setState((prevState: State) => {
+    this.setState<'viewMode'>((prevState: State) => {
```

Comment:
:information_source: It complains that `{ viewMode }` is missing `loading`, because I think it cannot automatically infer the generic type here, due to the `if` (it actually infers it to `never`).